packages/lib-core/src/types/store.ts

```diff
@@ -15,39 +15,35 @@ export enum PluginEventType {
    * - plugin failed to load, or there was an error while processing its extensions
    * - plugin was enabled or disabled (*)
    *
-   * (*) this triggers event {@link PluginEventType.ExtensionsChanged}
+   * (*) this may trigger event {@link PluginEventType.ExtensionsChanged}
    *
-   * Associated data getter: {@link PluginConsumer.getPluginInfo}
+   * Associated data getter: {@link PluginStoreInterface.getPluginInfo}
```

Comment:
@vojtechszocs sorry for the delay in reviewing. There is a lot here, but overall looks good.

Since we are changing several public APIs I'd like to test these changes in hac-core and associated plugins. 

Do you have areas you suspect that may be breaking for existing consumers I can focus on?

Reply:
> Since we are changing several public APIs I'd like to test these changes in hac-core and associated plugins.

:+1:

Reply:
> Do you have areas you suspect that may be breaking for existing consumers I can focus on?

@florkbr From a host app perspective, this should only impact how plugin dependencies are handled at runtime.

If an existing plugin has no dependencies, there should be no impact. You can try to modify an existing plugin and add some dependencies, then put `debugger` statements in the resolution logic to see that everything works as expected. You can test both "regular" and "fixed" dependency types, as documented in PR description.